django/contrib/admin/utils.py

```diff
@@ -297,10 +297,15 @@ def lookup_field(name, obj, model_admin=None):
             attr = getattr(model_admin, name)
             value = attr(obj)
         else:
-            attr = getattr(obj, name)
+            sentinel = object()
+            attr = getattr(obj, name, sentinel)
             if callable(attr):
                 value = attr()
             else:
+                if attr is sentinel:
+                    attr = obj
+                    for part in name.split(LOOKUP_SEP):
+                        attr = getattr(attr, part)
```

Comment:
Thanks, I didn't notice this. I re-appropriated your test. Not sure it is the best solution. I wanted to catch the `AttributeError` in the calling function as happens in other code calling this function, but there's a test guarding against this and I'm sure it's there for a reason. So I'm catching it here and returning early. I'm not sure it's cleanest solution but it's the best I could think of and the tests are passing. What do you think?

Reply:
Thank you @knyghty for the fix. I personally prefer to use the third param of `getattr` instead of catching the exception (old habit from when "exceptions were expensive"). So I would re-use the existing sentinel value to do:

```python
                         attr = getattr(attr, part, sentinel)                   
                         if attr is sentinel:                                   
                             return None, None, None                            
```

I've found another potential issue which I'll add as a separated comment.

Reply:
Yes, looks a bit cleaner, I've pushed this, though now I wonder which is actually faster these days 🤔

Reply:
Apparently, `getattr` with 3 params, by far!

```pycon
In [21]: def f1():
    ...:     if getattr("x", "foo", None) is None:
    ...:         pass
    ...: 

In [22]: def f2():
    ...:     try:
    ...:         "x".foo
    ...:     except AttributeError:
    ...:         pass
    ...: 

In [23]: def f3():
    ...:     try:
    ...:         getattr("x", "foo")
    ...:     except AttributeError:
    ...:         pass
    ...: 

In [24]: timeit.timeit(f1)
Out[24]: 0.1061433530121576

In [25]: timeit.timeit(f2)
Out[25]: 0.7686385459965095

In [26]: timeit.timeit(f3)
Out[26]: 0.7868501110060606
```